src/panels/config/integrations/integration-panels/zwave_js/zwave_js-node-config.ts

```diff
@@ -0,0 +1,425 @@
+              slot="dropdown-content"
+              .selected=${item.value}
+              attr-for-selected="value"
+              .key=${id}
+              .property=${item.property}
+              .propertyKey=${item.property_key}
+              @iron-select=${this._dropdownSelected}
+            >
+              ${Object.entries(item.metadata.states).map(
+                ([key, state]) => html`
+                  <paper-item .value=${key}>${state}</paper-item>
+                `
+              )}
+            </paper-listbox>
+          </paper-dropdown-menu>
+        </div>
+      `;
+    }
+
+    return html`${labelAndDescription}
+      <p>${item.value}</p>`;
+  }
+
+  private _isEnumeratedBool(item) {
+    // Some Z-Wave config values use a states list with two options where index 0 = Disabled and 1 = Enabled
+    // We want those to be considered boolean and show a toggle switch
+    const disabledStates = ["disable", "disabled"];
+    const enabledStates = ["enable", "enabled"];
+
+    if (item.configuration_value_type !== "enumerated") {
+      return false;
+    }
+    if (!("states" in item.metadata)) {
+      return false;
+    }
+    if (!(0 in item.metadata.states) || !(1 in item.metadata.states)) {
+      return false;
+    }
+    if (
+      disabledStates.includes(item.metadata.states[0].toLowerCase()) &&
+      enabledStates.includes(item.metadata.states[1].toLowerCase())
+    ) {
+      return true;
+    }
+    return false;
+  }
+
+  private _switchToggled(ev) {
+    this._updateConfigParameter(ev.target, ev.target.checked ? 1 : 0);
+  }
+
+  private _dropdownSelected(ev) {
+    if (ev.target === undefined || this._config![ev.target.key] === undefined) {
+      return;
+    }
+    if (this._config![ev.target.key].value === ev.target.selected) {
+      return;
+    }
+
+    this._updateConfigParameter(ev.target, parseInt(ev.target.selected));
+  }
+
+  private debouncedUpdate = debounce((target) => {
+    const value = parseInt(target.value);
+    this._config![target.key].value = value;
+
+    this._updateConfigParameter(target, value);
+  }, 1000);
+
+  private _numericInputChanged(ev) {
+    if (ev.target === undefined || this._config![ev.target.key] === undefined) {
+      return;
+    }
+    if (this._config![ev.target.key].value === parseInt(ev.target.value)) {
+      return;
+    }
+    this.debouncedUpdate(ev.target);
+  }
+
+  private _updateConfigParameter(target, value) {
+    const data: ZWaveJSSetConfigParamData = {
+      type: "zwave_js/set_config_parameter",
+      entry_id: this.configEntryId!,
+      node_id: this.nodeId!,
+      property: target.property,
+      value: value,
+    };
+
+    if (target.propertyKey !== null) {
+      data.property_key = target.propertyKey;
+    }
+
+    this.hass.callWS(data);
+
+    this._config![target.key].value = value;
+  }
+
+  private async _fetchData() {
+    if (!this.configEntryId) {
+      return;
+    }
+
+    this._device = this._deviceRegistryEntries
```

Comment:
`this._deviceRegistryEntries` might not be initialized yet ? 

Also, it's easier to memoize this. That way it's always in sync with the dev entries.

```ts
const getDevice = memoizeOne((entries, deviceId) => entries.find(…));
```

Reply:
There is no need to update it in this method, instead make it a getter. That way it's just calculated when it's needed and it can never get out of sync.

```
private _device(): DeviceRegistryEntry | undefined {
  return this._deviceRegistryEntries
      ? getDevice(this._deviceRegistryEntries, this.deviceId)
      : undefined;
}
```

Reply:
The 2nd comment wasn't addressed yet.